### ISSUE TITLE
[FEAT][ELEMENTWISE] UnaryKernel template ops (30 ops)

### ISSUE THREAD

**Issue by lcy-seso** [tile-ai/TileOPs#437]
## 1. Feature Scope

30 UnaryKernel subclasses. Each is 3-5 lines defining only `op_func(x)`. `UnaryKernel` base class from #436 handles strategy/config/forward.

| Category | Ops | Count |
|----------|-----|-------|
| unary_math | exp, log, sqrt, rsqrt, abs, neg, reciprocal, sign, sin, cos, floor, ceil, round, trunc, erf, log1p, expm1 | 17 |
| activation | gelu, silu, sigmoid, tanh, hardswish, hardsigmoid, mish, selu | 8 |
| logical | logical_not | 1 |
| bitwise | bitwise_not | 1 |
| special | isnan, isinf, isfinite | 3 |

**Excluded**: relu (#436), leaky_relu/elu/hardtanh/softplus/prelu (#439).

### Files

| File | Action |
|------|--------|
| `tileops/kernels/elementwise.py` | Add 30 kernel subclasses |
| `tileops/ops/elementwise.py` | Add 30 op subclasses |
| `tests/ops/test_unary_math.py` | New — 17 ops |
| `tests/ops/test_activation.py` | Append 8 ops to existing |
| `tests/ops/test_logical.py` | New — logical_not |
| `tests/ops/test_bitwise.py` | New — bitwise_not |
| `tests/ops/test_special_elementwise.py` | New — isnan, isinf, isfinite |

### Subclass Pattern

```python
class ExpKernel(UnaryKernel):
    @staticmethod
    def op_func(x):
        return T.exp(x)

class ExpOp(UnaryOp):
    _op_name = "exp"
    kernel_cls = ExpKernel
```

No custom `__init__` or `default_config`. All 30 ops follow this exact pattern.

### op_func Reference

| Kernel | `op_func(x)` return | Notes |
|--------|---------------------|-------|
| Exp | `T.exp(x)` | |
| Log | `T.log(x)` | |
| Sqrt | `T.sqrt(x)` | |
| Rsqrt | `T.rsqrt(x)` | |
| Abs | `T.abs(x)` | |
| Neg | `-x` | |
| Reciprocal | `1.0 / x` | |
| Sign | `T.if_then_else(x > 0, 1.0, T.if_then_else(x < 0, -1.0, 0.0))` | No `T.sign` builtin |
| Sin | `T.sin(x)` | |
| Cos | `T.cos(x)` | |
| Floor | `T.floor(x)` | |
| Ceil | `T.ceil(x)` | |
| Round | `T.round(x)` | |
| Trunc | `T.trunc(x)` | Fallback: `T.if_then_else(x >= 0, T.floor(x), T.ceil(x))` |
| Erf | `T.erf(x)` | |
| Log1p | `T.log(x + 1.0)` | No `T.log1p`; precision risk near 0 |
| Expm1 | `T.exp(x) - 1.0` | No `T.expm1`; precision risk near 0 |
| Gelu | `x * 0.5 * (1.0 + T.erf(x * 0.7071067811865476))` | |
| Silu | `x * T.sigmoid(x)` | Fallback: `x / (1.0 + T.exp(-x))` |
| Sigmoid | `T.sigmoid(x)` | Fallback: `1.0 / (1.0 + T.exp(-x))` |
| Tanh | `T.tanh(x)` | |
| Hardswish | `x * T.min(T.max(x + 3.0, 0.0), 6.0) / 6.0` | |
| Hardsigmoid | `T.min(T.max(x / 6.0 + 0.5, 0.0), 1.0)` | Fixed constants |
| Mish | `x * T.tanh(T.log(1.0 + T.exp(x)))` | |
| Selu | `scale * T.if_then_else(x > 0, x, alpha * (T.exp(x) - 1.0))` | alpha=1.6732632423543772, scale=1.0507009873554805 |
| LogicalNot | `x == 0` | Bool output |
| BitwiseNot | `~x` | Integer input |
| Isnan | `x != x` | Bool output |
| Isinf | `T.isinf(x)` | Fallback: `T.abs(x) == T.infinity()` |
| Isfinite | `(x == x) & (~T.isinf(x))` | Bool output |

If a builtin exists, prefer it over the composite fallback.

### L1 — Correctness (30 cases)

All 30 ops, fp16, 1M elements, same-shape, default strategy. Reference: PyTorch equivalent in fp32 then cast.

| Test File | Ops | PyTorch Reference |
|-----------|-----|-------------------|
| test_unary_math | exp...expm1 (17) | `torch.exp`, `torch.log`, ..., `torch.expm1` |
| test_activation | gelu...selu (8) | `F.gelu`, `F.silu`, `torch.sigmoid`, `torch.tanh`, `F.hardswish`, `F.hardsigmoid`, `F.mish`, `F.selu` |
| test_logical | logical_not (1) | `torch.logical_not` |
| test_bitwise | bitwise_not (1) | `torch.bitwise_not` |
| test_special | isnan, isinf, isfinite (3) | `torch.isnan`, `torch.isinf`, `torch.isfinite` |

Input domain: `randn` for most; `abs(randn) + eps` for log/sqrt/rsqrt.

### L4 — Edge Cases (10 cases, fp32, 4K)

| Op | Input | Expected |
|----|-------|----------|
| sqrt | `[-1, 0, 1e-38, 1]` | NaN, 0, ~1e-19, 1 |
| rsqrt | `[-1, 0, 1e-38, 1]` | NaN, Inf, ~1e19, 1 |
| log | `[-1, 0, 1e-38, 1]` | NaN, -Inf, ~-87.3, 0 |
| log1p | `[-2, -1, 0, 1e-7]` | NaN, -Inf, 0, ~1e-7 |
| exp | `[0, 88.8, -88.8, 200]` | 1, ~large, ~0, Inf |
| expm1 | `[0, 88.8, -88.8, 1e-7]` | 0, ~large, -1, ~1e-7 |
| erf | `[0, 3, -3, 100]` | 0, ~1, ~-1, 1 |
| reciprocal | `[0, 1, -1, 1e-38]` | Inf, 1, -1, ~1e38 |
| sign | `[-5, 0, 3, NaN]` | -1, 0, 1, 0 (or NaN) |
| isnan/isinf/isfinite | `[1, NaN, Inf, -Inf]` | Match PyTorch bool output |

Tolerance: fp32 atol=1e-5 rtol=1e-5, fp16 atol=1e-3 rtol=1e-3.

## 4. Risks

| Risk | Mitigation |
|------|------------|
| TileLang missing builtins (trunc, sigmoid, isinf) | Use composite fallbacks; prefer builtins if they exist |
| Bool output ops (logical_not, isnan, isinf, isfinite) may need `OUTPUT_DTYPE` on UnaryKernel | Same pattern as BinaryKernel comparison ops; coordinate with #436 if template change needed |
| log1p/expm1 composite expressions lose precision near 0 | L4 edge case tests will catch; switch to builtins if available |

## 5. Acceptance Criteria

- [ ] 30 kernel + 30 op subclasses implemented
- [ ] `__all__` updated in both `elementwise.py`
- [ ] L1: 30 ops pass (fp16, 1M, same-shape)
- [ ] L4: 10 edge cases pass (fp32, 4K)
- [ ] `__init__.py` not modified (concrete ops not re-exported)
